014-quarkus-panache-with-transactions-xa/src/test/java/io/quarkus/qe/dbpool/AgroalPoolTest.java

```diff
@@ -0,0 +1,148 @@
+import org.jboss.logging.Logger;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import io.quarkus.qe.containers.MysqlTestProfile;
+import io.quarkus.qe.rest.data.UserRepository;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.tuples.Tuple2;
+
+@QuarkusTest
+@TestProfile(MysqlTestProfile.class)
+@TestMethodOrder(OrderAnnotation.class)
+public class AgroalPoolTest {
+
+    private static final Logger LOG = Logger.getLogger(AgroalPoolTest.class);
+
+    @Inject
+    EntityManager em;
+
+    @Inject
+    UserRepository users;
+
+    static final int TIMEOUT_SEC = 60;
+
+    @ConfigProperty(name = "quarkus.datasource.jdbc.idle-removal-interval")
+    String idleSec;
+
+    @ConfigProperty(name = "quarkus.datasource.jdbc.max-size")
+    int datasourceMaxSize;
+
+    @ConfigProperty(name = "quarkus.datasource.jdbc.min-size")
+    int datasourceMinSize;
+
+    @Test
+    @Order(1)
+    public void agroalCheckIdleTimeout() throws InterruptedException {
+        makeApplicationQuery();
+        Thread.sleep(3000);
+        assertEquals(1, activeConnections(), "Expected " + datasourceMinSize + " active connections");
+    }
+
+    @Test
+    @Order(2)
+    public void agroalCheckDbPoolTurnover() throws InterruptedException {
+        final int events = 2500;
+        final AtomicInteger max = new AtomicInteger(0);
+        CountDownLatch completed = new CountDownLatch(events);
+
+        Multi<Tuple2<Long, Long>> combined = Multi.createBy().combining()
+                .streams(makeApplicationQueryAsync(events), activeConnectionsAsync(events)).asTuple();
+
+        combined.subscribe().with(t -> {
+            Long activeCon = t.getItem2();
+            if (max.intValue() < activeCon) {
+                max.set(activeCon.intValue());
+            }
+            assertTrue(datasourceMaxSize >= activeCon, "More mysql SQL connections than pool max-size");
+            assertTrue(datasourceMinSize <= activeCon, "Less mysql SQL connections than pool min-size");
+            completed.countDown();
+        });
+
+        LOG.info("Max connection requested: " + max);
+        completed.await(TIMEOUT_SEC, TimeUnit.SECONDS);
+        assertEquals(0, completed.getCount(), String.format("Missing %d events.", events - completed.getCount()));
+    }
+
+    @RepeatedTest(10)
+    @Order(3)
+    @Disabled("https://github.com/quarkusio/quarkus/issues/17141")
+    public void agroalCheckBorderConditionBetweenIdleAndGetConnection() throws InterruptedException {
+        final int concurrentRequestAmount = 2;
+        CountDownLatch completed = new CountDownLatch(concurrentRequestAmount);
+        LOG.info("######################");
```

Comment:
Consider adding a prefix to set it apart from identical log entry in `agroalConcurrentLoadTest`.